src/components/ColorPickerContainer.ts

```diff
@@ -0,0 +1,286 @@
+import { Input } from "./Input";
+import { Button } from "./Button";
+import { Label } from "./Label";
+
+interface WrapperProps {
+    class: string;
+    mxObject?: mendix.lib.MxObject;
+    mxform: mxui.lib.form._FormBase;
+    style: string;
+    readOnly: boolean;
+    friendlyId: string;
+}
+
+interface Nanoflow {
+    nanoflow: object[];
+    paramsSpec: { Progress: string };
+}
+
+export interface ColorPickerContainerProps extends WrapperProps {
+    colorAttribute: string;
+    editable: "default" | "never";
+    format: Format;
+    mode: Mode;
+    type: PickerType;
+    label: string;
+    labelWidth: number;
+    onChangeEvent: onChange;
+    onChangeMicroflow: string;
+    onChangePage: string;
+    onChangeNanoflow: Nanoflow;
+    openPageLocation: "content" | "popup" | "modal";
+    defaultColors: string[];
+}
+
+interface ColorPickerContainerState {
+    color: string;
+    alertMessage?: string;
+    displayColorPicker: boolean;
+}
+
+type Format = "hex" | "rgb" | "rgba";
+type onChange = "doNothing" | "showPage" | "callMicroflow" | "callNanoflow";
+
+export default class ColorPickerContainer extends Component<ColorPickerContainerProps, ColorPickerContainerState> {
+    private subscriptionHandles: number[] = [];
+    private disabled = false;
+    private previousColor = "";
+    private colorChanged = false;
+
+    constructor(props: ColorPickerContainerProps) {
+        super(props);
+
+        this.state = {
+            alertMessage: "",
+            color: "",
+            displayColorPicker: false
+        };
+    }
+
+    render() {
+        const { mxObject, readOnly, colorAttribute, format, type } = this.props;
+        this.disabled = this.props.editable !== "default"
+            || ((!mxObject || readOnly || !!(colorAttribute && mxObject.isReadonlyAttr(colorAttribute))
+            || (type === "alpha" && format !== "rgba")));
+
+        const maxLabelWidth = 11;
+        if (this.props.label.trim()) {
+            return createElement(Label, {
+                className: this.props.class,
+                label: this.props.label,
+                style: ColorPickerContainer.parseStyle(this.props.style),
+                weight: this.props.labelWidth > maxLabelWidth ? 6 : this.props.labelWidth
+            }, this.renderColorPicker(true));
+        }
+
+        return this.renderColorPicker();
+    }
+
+    componentWillReceiveProps(newProps: ColorPickerContainerProps) {
+        this.resetSubscriptions(newProps.mxObject);
+        this.setState(this.validateColor(this.getValue(newProps.mxObject)));
+    }
+
+    componentWillUnmount() {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+    }
+
+    private renderColorPicker(hasLabel = false) {
+        const alertMessage = this.state.alertMessage || ColorPickerContainer.validateProps(this.props);
+
+        return createElement(ColorPicker, {
+            alertMessage,
+            className: !hasLabel ? this.props.class : undefined,
```

Comment:
why not use the class when using label, please check core input widget behaviour

Reply:
The label becomes the parent and so the class has to be applied to the parent